lib/ansible/module_utils/facts/system/pkg_mgr.py

```diff
@@ -78,10 +78,11 @@ def collect(self, module=None, collected_facts=None):
             if os.path.exists(pkg['path']):
                 pkg_mgr_name = pkg['name']
 
-        if pkg_mgr_name == 'apt' and \
-                os.path.exists('/usr/bin/rpm') and \
-                not os.path.exists('/usr/bin/dpkg'):
-            pkg_mgr_name = 'apt_rpm'
+        if pkg_mgr_name == 'apt':
+            if os.path.exists('/etc/fedora-release'):
+                pkg_mgr_name = 'dnf'
```

Comment:
won't this break 'old' fedora that are 'pre dnf'?

Reply:
@bcoca to the best of my knowledge, Fedora that's pre-`dnf` don't have a version of python new enough to fall within the support matrix of current Ansible versions. Fedora has been `dnf` since Fedora 15 which was released 2011-05-24 .... so, yes? technically, but outside the scope of practical support (IMO anyway).

Reply:
ok, just wanted to make sure someone had looked it up

Reply:
@bcoca I'm incorrect, I just checked. Fedora 14 was the first release to ship with python 2.7. So my new question is: How far back in the history of EOL'd distros do we want to maintain compat for?

Reply:
I would use the word 'reasonable' but too many people make it unreasonable ....

fine with the change, the problem hits ancient version and this is mainly a desktop distro or 'cutting edge' so i expect almost no one will be hit by this.

/me now waits for thousands of users complaining

Reply:
@bcoca actually, I'll go ahead and add a check for version range that way we don't accidentally break anyone who happens to still have an old Fedora machine in their environment. (I'm sure that number is non-zero)

Reply:
it might be non-zero, but i suspect it is infinitely close to 0